code/llm/src/new_code/infer_embedding.py

```diff
@@ -1,28 +1,18 @@
-import pandas as pd
 import numpy as np
-import src.transformer
-from src.transformer.models import TransformerEncoder
-from src.new_code.load_data import CustomDataset
-from src.new_code.pretrain import read_hparams_from_file
-from src.new_code.utils import read_json, print_now
-from torch.utils.data import DataLoader
 import torch
 import json
-import pickle
-from torch.utils.data._utils.collate import default_collate
 import sys
+import logging 
 from tqdm import tqdm
 
-def custom_collate(batch):
-    collated_batch = {}
-    for key in batch[0].keys():
-        # Check if the key is 'sequence_id' and handle it separately
-        if key == 'sequence_id':
-            collated_batch[key] = [item[key] for item in batch]
-        else:
-            collated_batch[key] = default_collate([item[key] for item in batch])
+from src.transformer.models import TransformerEncoder
+from src.new_code.load_data import CustomIterableDataset
+from src.new_code.pretrain import read_hparams_from_file
+from src.new_code.pipeline import write_to_hdf5
+from src.new_code.utils import read_json, print_now
+from torch.utils.data import DataLoader
+
```

Comment:
The imports could be ordered better with `isort`.

Reply:
I agree. This, among other things, is on my todo-list in #42 . added isort to that issue